### PR TITLE
GangSTR Runner

### PR DESCRIPTION
**Pull request by hopedisastro**
Modified https://github.com/populationgenomics/sv-workflows/blob/main/str/str_iterative_eh_runner.py to create a runner for GangSTR. Pylint flagged that there was a lot of duplicate code, but I silenced this warning to pass the check.

### CODE REVIEW

File: str/str_iterative_gangstr_runner.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+@click.option('--project-id', help='project-id eg tob-wgs')
+# input sample ID
+@click.argument('external-wgs-ids', nargs=-1)
+@click.command()
+def main(
+    variant_catalog, project_id, external_wgs_ids: list[str]
+):  # pylint: disable=missing-function-docstring
+    # Initializing Batch
+    backend = hb.ServiceBackend(
+        billing_project=get_config()['hail']['billing_project'],
```
Comment (illusional) [inline str/str_iterative_gangstr_runner.py:55]:
```suggestion
        billing_project=config['hail']['billing_project'],
```

**Review by illusional — CHANGES_REQUESTED** [review]
Mostly minor changes, hopefully should be straight-forward to address :)

File: str/str_iterative_gangstr_runner.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        project_id, list(external_wgs_ids)
+    )
+    cpg_id_to_external_id = {
+        cpg_id: external_wgs_id
+        for external_wgs_id, cpg_id in external_id_to_cpg_id.items()
+    }
+    if project_id == 'tob-wgs':
+        ref_fasta = (
```
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply str/str_iterative_gangstr_runner.py:68]: That would make sense. (Though the discussion on what reference to record in the CRAM header is still ongoing:
https://github.com/populationgenomics/production-pipelines/issues/208)

File: str/str_iterative_gangstr_runner.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+# inputs:
+# variant catalog
+@click.option('--variant-catalog', help='Full path to Illumina Variants catalog')
+# input project ID
+@click.option('--dataset', help='dataset to operate on, eg: tob-wgs')
+# input sample ID
+@click.argument('external-wgs-ids', nargs=-1)
+@click.command()
+def main(
+    variant_catalog, project_id, external_wgs_ids: list[str]
```
Comment (illusional) [inline str/str_iterative_gangstr_runner.py:43]: This project_id and subsequent usages also need to be switched

File: str/str_iterative_gangstr_runner.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+SAMTOOLS_IMAGE = config['images']['samtools']
+GANGSTR_IMAGE = config['images']['gangstr']
+
+
+# inputs:
+# variant catalog
+@click.option('--variant-catalog', help='Full path to Illumina Variants catalog')
+# input project ID
+@click.option('--dataset', help='dataset to operate on, eg: tob-wgs')
```
Comment (illusional) [inline str/str_iterative_gangstr_runner.py:38]: Optional: You could make the default here the default from the config you had before. So running the script without a dataset param automatically uses the dataset provided in the config?

**Review by illusional — APPROVED** [review]
One small rename of the project_id is required, but otherwise looks good to me

**Pull request merged**